lib/firehose/server/message_buffer.rb

```diff
@@ -12,27 +12,36 @@ class MessageBuffer
       def initialize(message_list, channel_sequence, consumer_sequence = nil)
         @message_list = message_list
         @channel_sequence = channel_sequence.to_i
-        @consumer_sequence = consumer_sequence.to_i
+
+        if consumer_sequence
+          @consumer_sequence = consumer_sequence.to_i
+        else
+          @consumer_sequence = @message_list.size - 1
+        end
       end
 
       def consumed_messages
-        messages.take(offset)
+        messages.take consumed_message_count
       end
 
       def remaining_messages
-        messages.drop(offset)
+        messages.drop consumed_message_count
       end
 
-      def offset
-        # Somehow the sequence is negative. Consumer needs all messages
-        if @consumer_sequence < 0
+      def consumed_message_count
+        # Consumer is starting at 0 or somehow the value is negative. Start the consumer off
+        # with the lastest message.
+        if @message_list.empty?
```

Comment:
I don't see how `@message_list.empty?` is relevant here. The comment is correct, but the code here doesn't line up with the comment.